Binary files bfgminer-3.10.10.orig/api-example.o and bfgminer-3.10.10/api-example.o differ
Binary files bfgminer-3.10.10.orig/bfgminer and bfgminer-3.10.10/bfgminer differ
Binary files bfgminer-3.10.10.orig/bfgminer-adl.o and bfgminer-3.10.10/bfgminer-adl.o differ
Binary files bfgminer-3.10.10.orig/bfgminer-api.o and bfgminer-3.10.10/bfgminer-api.o differ
Binary files bfgminer-3.10.10.orig/bfgminer-boinc_wrapper.o and bfgminer-3.10.10/bfgminer-boinc_wrapper.o differ
Binary files bfgminer-3.10.10.orig/bfgminer-deviceapi.o and bfgminer-3.10.10/bfgminer-deviceapi.o differ
Binary files bfgminer-3.10.10.orig/bfgminer-driver-opencl.o and bfgminer-3.10.10/bfgminer-driver-opencl.o differ
Binary files bfgminer-3.10.10.orig/bfgminer-findnonce.o and bfgminer-3.10.10/bfgminer-findnonce.o differ
Binary files bfgminer-3.10.10.orig/bfgminer-logging.o and bfgminer-3.10.10/bfgminer-logging.o differ
Binary files bfgminer-3.10.10.orig/bfgminer-miner.o and bfgminer-3.10.10/bfgminer-miner.o differ
Binary files bfgminer-3.10.10.orig/bfgminer-ocl.o and bfgminer-3.10.10/bfgminer-ocl.o differ
Binary files bfgminer-3.10.10.orig/bfgminer-rpc and bfgminer-3.10.10/bfgminer-rpc differ
Binary files bfgminer-3.10.10.orig/bfgminer-scrypt.o and bfgminer-3.10.10/bfgminer-scrypt.o differ
Binary files bfgminer-3.10.10.orig/bfgminer-sha2.o and bfgminer-3.10.10/bfgminer-sha2.o differ
Binary files bfgminer-3.10.10.orig/bfgminer-util.o and bfgminer-3.10.10/bfgminer-util.o differ
diff -urN -x result.txt -x stderr.txt -x configure -x config.status -x config.log -x aclocal.m4 -x Makefile -x Makefile.in bfgminer-3.10.10.orig/boinc_wrapper.cpp bfgminer-3.10.10/boinc_wrapper.cpp
--- bfgminer-3.10.10.orig/boinc_wrapper.cpp	1969-12-31 16:00:00.000000000 -0800
+++ bfgminer-3.10.10/boinc_wrapper.cpp	2016-01-13 14:02:57.606572114 -0800
@@ -0,0 +1,60 @@
+
+#include "boinc_wrapper.h"
+
+#include "boinc_api.h"
+#include <filesys.h>
+
+BOINC_OPTIONS options;
+
+void bw_boinc_begin_critical_section()
+{
+	bw_boinc_begin_critical_section();
+}
+
+void bw_boinc_end_critical_section()
+{
+	boinc_end_critical_section();
+}
+
+void bw_boinc_fraction_done(double x)
+{
+	bw_boinc_fraction_done(x);
+}
+
+int bw_boinc_get_status_suspended()
+{
+	BOINC_STATUS bs;
+	boinc_get_status(&bs);
+	return bs.suspended;
+}
+
+void bw_boinc_init()
+{
+	boinc_init();
+}
+
+void bw_boinc_finish(int x)
+{
+	boinc_finish(x);
+}
+
+void bw_boinc_init_options()
+{
+        options.normal_thread_priority = true;
+	boinc_init_options(&options);
+}
+
+void bw_boinc_show_message(int status, const char *msg, int is_notice)
+{
+	;//boinc_finish_message(status, msg, is_notice);
+}
+
+FILE* bw_boinc_fopen(char *path, char *mode)
+{
+	boinc_fopen(path, mode);
+}
+
+void bw_boinc_fclose(FILE *file)
+{
+	fclose(file);
+}
diff -urN -x result.txt -x stderr.txt -x configure -x config.status -x config.log -x aclocal.m4 -x Makefile -x Makefile.in bfgminer-3.10.10.orig/boinc_wrapper.h bfgminer-3.10.10/boinc_wrapper.h
--- bfgminer-3.10.10.orig/boinc_wrapper.h	1969-12-31 16:00:00.000000000 -0800
+++ bfgminer-3.10.10/boinc_wrapper.h	2016-01-13 14:00:32.120793203 -0800
@@ -0,0 +1,27 @@
+#ifndef BOINC_WRAPPER_H
+#define BOINC_WRAPPER_H
+#include <stdio.h>
+
+
+//The Boinc API isn't C API native bro.
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+void bw_boinc_begin_critical_section();
+void bw_boinc_end_critical_section();
+void bw_boinc_fraction_done(double x);
+int bw_boinc_get_status_suspended();
+void bw_boinc_init();
+void bw_boinc_finish(int x);
+void bw_boinc_init_options();
+void bw_boinc_show_message(int status, const char *msg, int is_notice);
+FILE* bw_boinc_fopen(char *path, char *mode);
+void bw_boinc_fclose(FILE *file);
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif
Binary files bfgminer-3.10.10.orig/ccan/libccan.a and bfgminer-3.10.10/ccan/libccan.a differ
Binary files bfgminer-3.10.10.orig/ccan/opt/libccan_a-helpers.o and bfgminer-3.10.10/ccan/opt/libccan_a-helpers.o differ
Binary files bfgminer-3.10.10.orig/ccan/opt/libccan_a-opt.o and bfgminer-3.10.10/ccan/opt/libccan_a-opt.o differ
Binary files bfgminer-3.10.10.orig/ccan/opt/libccan_a-parse.o and bfgminer-3.10.10/ccan/opt/libccan_a-parse.o differ
Binary files bfgminer-3.10.10.orig/ccan/opt/libccan_a-usage.o and bfgminer-3.10.10/ccan/opt/libccan_a-usage.o differ
Binary files bfgminer-3.10.10.orig/lib/dummy.o and bfgminer-3.10.10/lib/dummy.o differ
Binary files bfgminer-3.10.10.orig/lib/libgnu.a and bfgminer-3.10.10/lib/libgnu.a differ
diff -urN -x result.txt -x stderr.txt -x configure -x config.status -x config.log -x aclocal.m4 -x Makefile -x Makefile.in bfgminer-3.10.10.orig/miner.c bfgminer-3.10.10/miner.c
--- bfgminer-3.10.10.orig/miner.c	2016-01-13 12:46:58.209566700 -0800
+++ bfgminer-3.10.10/miner.c	2016-01-13 13:36:53.889151864 -0800
@@ -110,7 +110,6 @@
 };
 
 #ifdef USE_BOINC
-BW_BOINC_OPTIONS options;
 int boinc_success = 0;
 #endif
 static char packagename[256];
@@ -11368,7 +11367,7 @@
 #ifdef USE_BOINC
 	bw_boinc_init();
 
-	bw_boinc_init_options(&options);
+	bw_boinc_init_options();
 #endif
 
 #ifdef WIN32
diff -urN -x result.txt -x stderr.txt -x configure -x config.status -x config.log -x aclocal.m4 -x Makefile -x Makefile.in bfgminer-3.10.10.orig/miner.h bfgminer-3.10.10/miner.h
--- bfgminer-3.10.10.orig/miner.h	2016-01-13 12:46:58.213566853 -0800
+++ bfgminer-3.10.10/miner.h	2016-01-13 13:28:54.515515174 -0800
@@ -755,6 +755,11 @@
 
 #define flip32(dest_p, src_p) swap32yes(dest_p, src_p, 32 / 4)
 
+#ifdef USE_BOINC
+void boinc_quit_status(int status, const char *format, ...);
+void boinc_work_success();
+#endif
+
 #define WATCHDOG_INTERVAL  2
 extern void bfg_watchdog(struct cgpu_info *, struct timeval *tvp_now);
 
@@ -1560,4 +1565,5 @@
 extern struct api_data *api_add_diff(struct api_data *root, char *name, double *data, bool copy_data);
 extern struct api_data *api_add_json(struct api_data *root, char *name, json_t *data, bool copy_data);
 
+
 #endif /* __MINER_H__ */
